src/Event.hh

```diff
@@ -27,51 +27,57 @@ struct Event {
 class EventList {
 public:
   void create(std::string path) {
-    Event *event = update(path);
+    std::lock_guard<std::mutex> l(mMutex);
```

Comment:
The locking in this file might be a bit overkill, we probably don't really need it anymore as the real issue was in watcher reassigning and reading the events array/map at the same time